src/webgpu/shader/execution/expression/call/builtin/clamp.spec.ts

```diff
@@ -25,28 +25,81 @@ import { builtin } from './builtin.js';
 
 export const g = makeTestGroup(GPUTest);
 
-/** Generate set of clamp test cases from an ascending list of values */
-function generateTestCases(test_values: Array<Scalar>): Array<Case> {
+/**
+ * Calculates clamp using the min-max formula.
+ * clamp(e, f, g) = min(max(e, f), g)
+ *
+ * Operates on indices of an ascending sorted array, instead of the actual
+ * values to avoid rounding issues.
+ *
+ * @returns the index of the clamped value
+ */
+function calculateMinMaxClamp(ei: number, fi: number, gi: number): number {
+  let idx = ei >= fi ? ei : fi;
```

Comment:
I think you can do

      return Math.min(Math.max(ei,fi),gi)

Reply:
Done